### ISSUE TITLE
macOS support for 2.8 / Big Sur

### ISSUE THREAD

**Issue by devvmh** [mnemosyne-proj/mnemosyne#208]
Something is broken on the latest installer. I've been trying for a few months on and off to make it work. I figure I should start documenting what I'm doing in case someone else can solve it or contribute ideas.

Today I tried again with the latest software, with only minor patches to the instructions in the existing README.md.

```
(venv) MacBook-Pro:mnemosyne devin$ python3 --version
Python 3.9.8
(venv) MacBook-Pro:mnemosyne devin$ brew list --versions qt@5
qt@5 5.15.2_1
```

I needed to modify one line in the makefile:

```diff
        # Build the bundled app based on the specification file.
-       QT5DIR=/usr/local/opt/qt pyinstaller --log-level WARN mnemosyne.spec
+       QT5DIR=/usr/local/opt/qt5 pyinstaller --log-level WARN mnemosyne.spec^M
```

Output when running `make macos`: https://pastebin.com/Fvsj7m2U

The built app crashes immediately. If I manually execute `dist/Mnemosyne/Mnemosyne`, I get this output:

```
[30647] Failed to execute script 'pyi_rth_multiprocessing' due to unhandled exception: No module named '_socket'
[30647] Traceback:
Traceback (most recent call last):
  File "PyInstaller/hooks/rthooks/pyi_rth_multiprocessing.py", line 17, in <module>
  File "PyInstaller/loader/pyimod03_importers.py", line 546, in exec_module
  File "multiprocessing/__init__.py", line 16, in <module>
  File "PyInstaller/loader/pyimod03_importers.py", line 546, in exec_module
  File "multiprocessing/context.py", line 6, in <module>
  File "PyInstaller/loader/pyimod03_importers.py", line 546, in exec_module
  File "multiprocessing/reduction.py", line 16, in <module>
  File "PyInstaller/loader/pyimod03_importers.py", line 546, in exec_module
  File "socket.py", line 51, in <module>
ModuleNotFoundError: No module named '_socket'
```

If I copy all missing dylib files with `cp  /usr/local/Cellar/python\@3.9/3.9.8/Frameworks/Python.framework/Versions/3.9/lib/python3.9/lib-dynload/* .`, then execute again, the output is: 

```
Traceback (innermost last):
  File "mnemosyne/pyqt_ui/mnemosyne", line 278, in <module>
  File "mnemosyne/libmnemosyne/__init__.py", line 400, in initialise
  File "mnemosyne/libmnemosyne/__init__.py", line 465, in register_components
  File "importlib/__init__.py", line 127, in import_module
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "PyInstaller/loader/pyimod03_importers.py", line 546, in exec_module
  File "mnemosyne/pyqt_ui/qt_sync_server.py", line 15, in <module>
  File "PyInstaller/loader/pyimod03_importers.py", line 546, in exec_module
  File "mnemosyne/libmnemosyne/sync_server.py", line 9, in <module>
  File "PyInstaller/loader/pyimod03_importers.py", line 546, in exec_module
  File "argon2/__init__.py", line 1, in <module>
  File "PyInstaller/loader/pyimod03_importers.py", line 546, in exec_module
  File "argon2/low_level.py", line 13, in <module>
 ImportError: dlopen(/Users/devin/git/mnemosyne/dist/Mnemosyne/_cffi_backend.cpython-39-darwin.so, 2): Library not loaded: @loader_path/libffi.7.dylib
  Referenced from: /Users/devin/git/mnemosyne/dist/Mnemosyne/_cffi_backend.cpython-39-darwin.so
  Reason: image not found
```

Seems like I have two issues - one with loading the core dylib files, and another with ffi.

**Comment by pbienst** [comment]
Thanks for your efforts! It could be a good idea to get in touch with the PyInstaller people and post a bug report there...

**Comment by devvmh** [comment]
Seems like these have been fixed as of pyinstaller 5.1. I also was able to figure out codesigning. I'm uploading a 2.8 build now and will post a PR with the fixes today

**Comment by devvmh** [comment]
For posterity: unclear if this is actually working yet! Code signing seems to cause some issues so I want to do a bit more testing. I need a 2nd laptop with Big Sur on it to be able to test fully though.
